.gitignore

```diff
@@ -21,3 +20,7 @@ contrib/build/*/tmp/*
 .kube
 .var
 .vscode
+vendor/github.com/petar/GoLLRB/doc/*
+contrib/examples/consumer/Gopkg.lock
+contrib/examples/consumer
+contrib/examples/vendor/*
```

Comment:
besides gollrb, are these to just avoid dealing with contrib? dep automagically finds them or something?

Reply:
contrib/examples/consumer is something that I added to test a downstream consumer using dep to vendor the Service Catalog client library. `make test-dep` run dep to vendor our generated clientset, and ensures that the app can compile. I added those ignore lines so that after running that target, a bunch of generated test files don't show up when someone runs `git status`. 😀 

That test-dep target helps us sanity check that the constraints in our Gopkg.toml aren't missing anything with respect to downstream consumers. Otherwise, we wouldn't find out for example that we need to be pinning to a particular revision, or avoiding a bug with a semver range until someone reported it to us.

When we use dep internally when working on service catalog, dep takes into account our Gopkg.lock where all of the dependencies are locked to an exact revision. When someone vendors us using dep, only Gopkg.toml is used. That's why I was asking a bunch of questions in the sig meeting on Monday about why in our glide.yaml so many projects were pinned to an exact revision. I wanted to figure out how much we *had* to manually manage in the Gopkg.toml and what we could "get away with" letting dep deal with in the Gopkg.lock file.

Reply:
oh neat!